stock_reorder_forecast/models/res_partner.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# © 2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+'''
+Additional fields for the purchase proposal
+'''
+from openerp import fields, api, models
+
+
+class ResPartner(models.Model):
+
+    _inherit = 'res.partner'
+
+    @api.model
+    def sql_update_partner(self):
+        sql = """
+        UPDATE res_partner RP
```

Comment:
this is significantly more performant via sql.   ORM implementation would had been more readable but the time of the cronjob , especially on large db's would had been quit long. Definetley more performant. on a average DB (1000+ products) this fetches all the information we need in one pass in under 2 minutes. I would suppose the ORM version could be above 10 minutes. THe function might also be triggered manually.